scripts/test/Muon/elemental_analysis_2/EA_grouping_tab/elemental_analysis_grouping_tab_presenter_test.py

```diff
@@ -83,6 +83,35 @@ def test_clear(self):
 
         self.assertEqual(self.grouping_table_view.num_rows(), 0)
 
+    @mock.patch("Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_tab_model.EAGroupingTabModel.is_data_loaded")
+    @mock.patch("Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_tab_widget_presenter.EAGroupingTabPresenter."
+                "update_view_from_model")
+    @mock.patch("Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_tab_widget_presenter.EAGroupingTabPresenter."
+                "update_description_text")
+    @mock.patch("Muon.GUI.ElementalAnalysis2.grouping_widget.ea_grouping_tab_widget_presenter.EAGroupingTabPresenter."
+                "plot_default_groups")
+    def test_handle_data_loaded_when_data_loaded(self, mock_plot_default_groups, mock_update_text, mock_update_view,
```

Comment:
This may be a style choice but patching doesn't make much sense to do here since you have the presenter and model directly. So you could rewrite this test into something like:

```
def test_handle_data_loaded_when_data_loaded(self):  
  self.model.is_data_loaded = mock.Mock(return_value=True)     
  self.presenter.update_view_from_model = mock.Mock()     
  self.presenter.update_description_text = mock.Mock()     
  self.presenter.plot_default_groups = mock.Mock() 

  self.presenter.handle_new_data_loaded()      

  # Assert statements     
  self.presenter.update_view_from_model.assert_called_once()     
  self.presenter.update_description_text.assert_called_once()     
  self.presenter.plot_default_groups.assert_called_once()
```

This way it removes the direct use of paths and looks a little cleaner imo. This is something which could be done with all the patches